src/Nethermind/Nethermind.Optimism/OptimismBlockValidator.cs

```diff
@@ -36,14 +37,38 @@ public class OptimismBlockValidator(
     private const string NonNullWithdrawalsRootError =
         $"{nameof(BlockHeader.WithdrawalsRoot)} is not null";
 
-    protected override bool ValidateWithdrawals(Block block, IReleaseSpec spec, out string? error)
+    public override bool ValidateBodyAgainstHeader(BlockHeader header, BlockBody toBeValidated, out string? errorMessage)
     {
-        BlockHeader header = block.Header;
+        if (!ValidateTxRootMatchesTxs(header, toBeValidated, out Hash256? txRoot))
+        {
+            errorMessage = BlockErrorMessages.InvalidTxRoot(header.TxRoot!, txRoot);
+            return false;
+        }
 
+        if (!ValidateUnclesHashMatches(header, toBeValidated, out _))
+        {
+            errorMessage = BlockErrorMessages.InvalidUnclesHash;
+            return false;
+        }
+
+        if (!ValidateWithdrawals(header, toBeValidated, out errorMessage))
```

Comment:
Can this sequence be problematic in the future, if for any reason the `ValidateBodyAgainstHeader` is changed a method is added but omitted here? Probably, unlikely.